extension/src/extension.ts

```diff
@@ -0,0 +1,56 @@
+/* The extension */
+import { EmulatorState, EmulatorController } from './emulator/emulator-controller'
+import { CommandController } from './commands/command-controller'
+import { refreshCodeLenses } from './utils/utils'
+import { Account } from './emulator/account'
+import { UIController } from './ui/ui-controller'
+import { ExtensionContext } from 'vscode'
+
+// The container for all data relevant to the extension.
+export class Extension {
+  // The extension singleton
+  static #instance: Extension
+
+  static initialize (ctx: ExtensionContext): Extension {
+    Extension.#instance = new Extension(ctx)
+    return Extension.#instance
+  }
+
+  ctx: ExtensionContext
+  UICtrl: UIController
```

Comment:
```suggestion
  uiCtrl: UIController
```

Small nitpick, feel free to keep as is